docs/features/highlight.md

```diff
@@ -0,0 +1,138 @@
+
+## Demo
+
+{@snippet features/highlight}
+
+## Configuring the highlight options
+
+It is possible to configure which highlight options are supported by the editor.
+You can use the {@link module:highlight/highlight~HighlightConfig#options `highlight.options`} configuration and define your own highlight styles.
+
+For example, the following editor supports only two styles (a green marker and a blue pen):
+
+```js
+ClassicEditor
+	.create( document.querySelector( '#editor' ), {
+		highlight: {
+			options: [
+				{
+					model: 'greenMarker',
+					class: 'marker-green',
+					title: 'Green marker',
+					color: '#66ff00',
+					type: 'marker'
+				},
+				{
+					model: 'bluePen',
+					class: 'pen-blue',
+					title: 'Blue pen',
+					color: '#0091ff',
+					type: 'pen'
+				}
+			]
+		},
+		toolbar: [
+			'headings', '|', 'bulletedList', 'numberedList', 'highlightDropdown', 'undo', 'redo'
+		]
+	} )
+	.then( ... )
+	.catch( ... );
+```
+
+{@snippet features/custom-highlight-options}
+
+Instead of using the (default) `highlightDropdown`, the feature also supports a configuration with separate buttons directly in the toolbar:
+
+```js
+ClassicEditor
+	.create( document.querySelector( '#editor' ), {
+		toolbar: {
+			items: [
+				'headings', '|', 'highlight:marker', 'highlight:greenMarker',
+				'highlight:pinkMarker', 'highlight:bluePen',
+				'highlight:redPen', 'removeHighlight', 'undo', 'redo'
+			]
+		}
+	} )
+	.then( ... )
+	.catch( ... );
+```
+
+{@snippet features/highlight-buttons}
+
+## Installation
+
+To add this feature to your editor install the [`@ckeditor/ckeditor5-highlight`](https://www.npmjs.com/package/@ckeditor/ckeditor5-highlight) package:
+
+```
+npm install --save @ckeditor/ckeditor5-highlight
+```
+
+And add it to your plugin list and the toolbar configuration:
+
+```js
+import Highlight from '@ckeditor/ckeditor5-highlight/src/highlight';
+
+ClassicEditor
+	.create( document.querySelector( '#editor' ), {
+		plugins: [ Highlight, ... ],
+		toolbar: [ 'highlightDropdown', ... ]
+	} )
+	.then( ... )
+	.catch( ... );
+```
+
+<info-box info>
+	Read more about {@link builds/guides/development/installing-plugins installing plugins}.
+</info-box>
+
+## Common API
+
+The {@link module:highlight/highlight~Highlight} plugin registers:
+
+* The `'highlightDropdown'` dropdown,
+* The {@link module:highlight/highlightcommand~HighlightCommand `'highlight'`} command.
+
+	The number of options and their names correspond to the {@link module:highlight/highlight~HighlightConfig#options `highlight.options`} configuration option.
+
+	You can change the highlight of the current selection by executing the command with a desired value:
+
+	```js
+	editor.execute( 'highlight', { value: 'marker' } );
+	```
+
+	The `value` corresponds to the `model` property in configuration object. For the default configuration:
+	```js
+	highlight.options = [
```

Comment:
I think we need to work on the colors because AFAIR the has been chosen without any deeper research.

Reply:
I handled it in https://github.com/ckeditor/ckeditor5-highlight/pull/6/commits/870c78c6cee2febfe6fc8fd10dfc4fb6b99d461d.